napari/components/viewer_model.py

```diff
@@ -719,12 +720,22 @@ def add_labels(
             {'opaque', 'translucent', and 'additive'}.
         visible : bool
             Whether the layer visual is currently being displayed.
+        path : str or list of str
+            Path or list of paths to image data. Paths can be passed as strings
+            or `pathlib.Path` instances.
 
         Returns
         -------
         layer : :class:`napari.layers.Labels`
             The newly-created labels layer.
         """
+        if data is None and path is None:
```

Comment:
How do we feel about automatically creating a labels layer of the same shape and type (np, dask, zarr) as the currently selected layer in the viewer? Much like add_points. dask probably requires a lot of engineering, but zarr probably works out of the box and would allow labeling big image volumes.

Reply:
(Though I’m happy to leave this for a future pr)

Reply:
yes, we should do this (i know you requested in #470, but let's make it a new PR)